packages/core/test/features/st-var.spec.ts

```diff
@@ -1259,162 +1284,569 @@ describe(`features/st-var`, () => {
+                const { stylable, sheets } = testStylableCore(
+                    `
+                    :vars {
+                        map: st-map(first 1px,second 2px);
+                    }
 
-            const { meta } = sheets['/entry.st.css'];
+                    `
+                );
+
+                const meta = sheets['/entry.st.css'].meta;
+                const flattenStVars = stylable.stVar.flatten(meta);
+
+                expect(flattenStVars).to.eql([
+                    {
+                        symbol: 'map.first',
+                        value: '1px',
+                        path: ['map', 'first'],
+                    },
+                    {
+                        symbol: 'map.second',
+                        value: '2px',
+                        path: ['map', 'second'],
+                    },
+                ]);
+            });
+
+            it('should flat custom value st vars', () => {
+                const { stylable, sheets } = testStylableCore({
+                    '/entry.st.css': `
+                        @st-import [stBorder] from './st-border.js';
+
+                        :vars {
+                            border: stBorder(1px, solid, red);
+                        }
+                    `,
+                    '/st-border.js': stBorderDefinitionMock,
+                });
 
-            const computedVars = stylable.stVar.getComputed(meta);
+                const meta = sheets['/entry.st.css'].meta;
+                const flattenStVars = stylable.stVar.flatten(meta);
 
-            expect(Object.keys(computedVars)).to.eql(['validBefore', 'invalid', 'validAfter']);
-            expect(computedVars.validBefore).to.containSubset({
-                value: 'red',
-                input: undefined,
-                diagnostics: { reports: [] },
+                expect(flattenStVars).to.eql([
+                    {
+                        symbol: 'border',
+                        value: '1px solid red',
+                        path: ['border'],
+                    },
+                ]);
             });
-            expect(computedVars.validAfter).to.containSubset({
-                value: 'green',
-                input: undefined,
-                diagnostics: { reports: [] },
+            it('should flat nested st-array st vars', () => {
+                const { stylable, sheets } = testStylableCore(
+                    `
+                    :vars {
+                        nestedArray: st-array(
+                            red,
+                            st-array(red, green),
+                            st-map(color1 gold, color2 blue),
+                        );
+                    }
+
+                    `
+                );
+
+                const meta = sheets['/entry.st.css'].meta;
+                const flattenStVars = stylable.stVar.flatten(meta);
+
+                expect(flattenStVars).to.eql([
+                    {
+                        symbol: 'nestedArray[0]',
```

Comment:
It's might be problematic to stringify the path in some cases, it should be escaped when needed.